Fixes #130: Add tests to cover RAML v1.0.1 patch

postatum opened the pull request:
Fixes #130: Add tests to cover RAML v1.0.1 patch

Each commit description contains a link to related github issue from [RAML v1.0.1 patch](https://github.com/raml-org/raml-spec/milestone/7).

jstoiko commented on manifest.json:
```diff
@@ -1,6 +1,7 @@
 {
     "description": "RAML files listed in order corresponding RAML feature appears in RAML 1.0 spec",
     "filePaths": [
+        "tests/raml-1.0/Root/yaml-12-compliancy/valid.raml",
```
Is there some YAML <1.2 syntax we could add as "invalid" test case?

postatum replied:
Yes. Added.

jstoiko commented:
> Each commit description contains a link to related github issue from RAML v1.0.1 patch.

@postatum: can we have a bullet list of all GH v1.0.1 issues with an enumeration of each corresponding test and links to those tests. E.g. `- issue raml-org/raml-spec#123: test 1, test 2, test 3`

postatum commented:
@jstoiko 

Added tests for:
* https://github.com/raml-org/raml-spec/issues/719:
  * tests/raml-1.0/Fragments/extension-relative-include/extensions/invalid-paths-relative-to-base.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Fragments/extension-relative-include/extensions/invalid-paths-relative-to-base.raml)
  * tests/raml-1.0/Fragments/extension-relative-include/extensions/valid-paths-relative-to-extension.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Fragments/extension-relative-include/extensions/valid-paths-relative-to-extension.raml)
  * tests/raml-1.0/Overlays/relative-include-lib/overlays/invalid-paths-relative-to-base.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Overlays/relative-include-lib/overlays/invalid-paths-relative-to-base.raml)
  * tests/raml-1.0/Overlays/relative-include-lib/overlays/valid-paths-relative-to-overlay.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Overlays/relative-include-lib/overlays/valid-paths-relative-to-overlay.raml)
* https://github.com/raml-org/raml-spec/issues/718:
  * tests/raml-1.0/Libraries/used-by-annotationtypedeclaration/valid-fragment.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-annotationtypedeclaration/valid-fragment.raml)
  * tests/raml-1.0/Libraries/used-by-annotationtypedeclaration/valid-spec.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-annotationtypedeclaration/valid-spec.raml)
  * tests/raml-1.0/Libraries/used-by-api-1/invalid-uses-inexisting-lib.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-api-1/invalid-uses-inexisting-lib.raml)
  * tests/raml-1.0/Libraries/used-by-api-1/valid.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-api-1/valid.raml)
  * tests/raml-1.0/Libraries/used-by-api-2/invalid-uses-non-lib.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-api-2/invalid-uses-non-lib.raml)
  * tests/raml-1.0/Libraries/uses-02/valid-indirect-use.raml → tests/raml-1.0/Libraries/used-by-api-2/valid-indirect-use.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-api-2/valid-indirect-use.raml)
  * tests/raml-1.0/Libraries/used-by-datatype/valid-spec.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-datatype/valid-spec.raml)
  * tests/raml-1.0/Libraries/used-by-documentationitem/valid-fragment.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-documentationitem/valid-fragment.raml)
  * tests/raml-1.0/Libraries/used-by-documentationitem/valid-spec.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-documentationitem/valid-spec.raml)
  * tests/raml-1.0/Libraries/used-by-extension/valid-fragment.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-extension/valid-fragment.raml)
  * tests/raml-1.0/Libraries/used-by-extension/valid-spec.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-extension/valid-spec.raml)
  * tests/raml-1.0/Libraries/used-by-library/valid-fragment.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-library/valid-fragment.raml)
  * tests/raml-1.0/Libraries/used-by-library/valid-spec.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-library/valid-spec.raml)
  * tests/raml-1.0/Libraries/used-by-namedexample/valid-fragment.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-namedexample/valid-fragment.raml)
  * tests/raml-1.0/Libraries/used-by-namedexample/valid-spec.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-namedexample/valid-spec.raml)
  * tests/raml-1.0/Libraries/used-by-overlay/valid-fragment.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-overlay/valid-fragment.raml)
  * tests/raml-1.0/Libraries/used-by-overlay/valid-spec.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-overlay/valid-spec.raml)
  * tests/raml-1.0/Libraries/used-by-resourcetype/valid-fragment.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-resourcetype/valid-fragment.raml)
  * tests/raml-1.0/Libraries/used-by-resourcetype/valid-spec.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-resourcetype/valid-spec.raml)
  * tests/raml-1.0/Libraries/used-by-securityscheme/valid-fragment.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-securityscheme/valid-fragment.raml)
  * tests/raml-1.0/Libraries/used-by-securityscheme/valid-spec.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-securityscheme/valid-spec.raml)
  * tests/raml-1.0/Libraries/used-by-trait/valid-fragment.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-trait/valid-fragment.raml)
  * tests/raml-1.0/Libraries/used-by-trait/valid-spec.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-trait/valid-spec.raml)
* https://github.com/raml-org/raml-spec/issues/715:
  * tests/raml-1.0/Root/yaml-12-compliancy/valid.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Root/yaml-12-compliancy/valid.raml)
* https://github.com/raml-org/raml-spec/issues/714:
  * tests/raml-1.0/Types/string-pattern-validation/invalid-does-not-match-pattern.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/string-pattern-validation/invalid-does-not-match-pattern.raml)
  * tests/raml-1.0/Types/string-pattern-validation/valid-example-matches-pattern.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/string-pattern-validation/valid-example-matches-pattern.raml)
* https://github.com/raml-org/raml-spec/issues/712:
  * tests/raml-1.0/Types/Type Expressions/object-union-with-enum/invalid-incorrect-enum-types.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/Type%20Expressions/object-union-with-enum/invalid-incorrect-enum-types.raml)
  * tests/raml-1.0/Types/Type Expressions/object-union-with-enum/valid-correct-enum-types.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/Type%20Expressions/object-union-with-enum/valid-correct-enum-types.raml)
  * tests/raml-1.0/Types/Type Expressions/scalar-union-with-enum/invalid-incorrect-enum-types.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/Type%20Expressions/scalar-union-with-enum/invalid-incorrect-enum-types.raml)
  * tests/raml-1.0/Types/Type Expressions/scalar-union-with-enum/valid-correct-enum-types.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/Type%20Expressions/scalar-union-with-enum/valid-correct-enum-types.raml)
* https://github.com/raml-org/raml-spec/issues/711:
  * tests/raml-1.0/Types/Type Expressions/union-with-facets/invalid-not-supported-facet.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/Type%20Expressions/union-with-facets/invalid-not-supported-facet.raml)
  * tests/raml-1.0/Types/Type Expressions/union-with-facets/valid-custom-facet.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/Type%20Expressions/union-with-facets/valid-custom-facet.raml)
  * tests/raml-1.0/Types/Type Expressions/union-with-facets/valid-supported-facet.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/Type%20Expressions/union-with-facets/valid-supported-facet.raml)
* https://github.com/raml-org/raml-spec/issues/710:
  * tests/raml-1.0/Types/determine-default-types/valid-array.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/determine-default-types/valid-array.raml)
  * tests/raml-1.0/Types/determine-default-types/valid-file.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/determine-default-types/valid-file.raml)
  * tests/raml-1.0/Types/determine-default-types/valid-integer.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/determine-default-types/valid-integer.raml)
  * tests/raml-1.0/Types/determine-default-types/valid-number.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/determine-default-types/valid-number.raml)
  * tests/raml-1.0/Types/determine-default-types/valid-object.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/determine-default-types/valid-object.raml)
  * tests/raml-1.0/Types/determine-default-types/valid-request-body-media-type.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/determine-default-types/valid-request-body-media-type.raml)
  * tests/raml-1.0/Types/determine-default-types/valid-request-body.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/determine-default-types/valid-request-body.raml)
  * tests/raml-1.0/Types/determine-default-types/valid-response-body-media-type.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/determine-default-types/valid-response-body-media-type.raml)
  * tests/raml-1.0/Types/determine-default-types/valid-response-body.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/determine-default-types/valid-response-body.raml)
  * tests/raml-1.0/Types/determine-default-types/valid-string-type-not-specified.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/determine-default-types/valid-string-type-not-specified.raml)
  * tests/raml-1.0/Types/determine-default-types/valid-string-with-pattern.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/determine-default-types/valid-string-with-pattern.raml)
* https://github.com/raml-org/raml-spec/issues/709:
  * tests/raml-1.0/Examples/mix-json-xml/invalid-incorrect-examples.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/mix-json-xml/invalid-incorrect-examples.raml)
  * tests/raml-1.0/Examples/mix-json-xml/valid.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/mix-json-xml/valid.raml)
  * tests/raml-1.0/Examples/mix-raml-json/invalid-incorrect-examples.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/mix-raml-json/invalid-incorrect-examples.raml)
  * tests/raml-1.0/Examples/mix-raml-json/valid.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/mix-raml-json/valid.raml)
  * tests/raml-1.0/Examples/mix-raml-xml/invalid-incorrect-examples.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/mix-raml-xml/invalid-incorrect-examples.raml)
  * tests/raml-1.0/Examples/mix-raml-xml/valid.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/mix-raml-xml/valid.raml)
  * tests/raml-1.0/Examples/multiple-json-examples/invalid-example-structure.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/multiple-json-examples/invalid-example-structure.raml)
  * tests/raml-1.0/Examples/multiple-json-examples/valid.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/multiple-json-examples/valid.raml)
  * tests/raml-1.0/Examples/multiple-with-strict-parameter/invalid-strict-true.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/multiple-with-strict-parameter/invalid-strict-true.raml)
  * tests/raml-1.0/Examples/multiple-with-strict-parameter/valid-strict-false.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/multiple-with-strict-parameter/valid-strict-false.raml)
  * tests/raml-1.0/Examples/multiple-xml-examples/invalid-wrong-example-value.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/multiple-xml-examples/invalid-wrong-example-value.raml)
  * tests/raml-1.0/Examples/multiple-xml-examples/valid.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/multiple-xml-examples/valid.raml)
  * tests/raml-1.0/Examples/single-json-example/invalid-example-structure.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/single-json-example/invalid-example-structure.raml)
  * tests/raml-1.0/Examples/single-json-example/valid.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/single-json-example/valid.raml)
  * tests/raml-1.0/Examples/single-with-strict-parameter/invalid-strict-true.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/single-with-strict-parameter/invalid-strict-true.raml)
  * tests/raml-1.0/Examples/single-with-strict-parameter/valid-strict-false.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/single-with-strict-parameter/valid-strict-false.raml)
  * tests/raml-1.0/Examples/single-xml-example/invalid-wrong-example-value.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Examples/single-xml-example/invalid-wrong-example-value.raml)
* https://github.com/raml-org/raml-spec/issues/708:
  * tests/raml-1.0/Types/nilable-type/invalid-prefixed-used-in-array.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/nilable-type/invalid-prefixed-used-in-array.raml)
  * tests/raml-1.0/Types/nilable-type/invalid-prefixed-used-in-union-array.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/nilable-type/invalid-prefixed-used-in-union-array.raml)
  * tests/raml-1.0/Types/nilable-type/invalid-prefixed-used-in-union.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/nilable-type/invalid-prefixed-used-in-union.raml)
  * tests/raml-1.0/Types/nilable-type/valid-nil-union.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/nilable-type/valid-nil-union.raml)
  * tests/raml-1.0/Types/nilable-type/valid-question-mark-suffix.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/nilable-type/valid-question-mark-suffix.raml)
  * tests/raml-1.0/EdgeCases/type-with-trailing-question-mark/ [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/EdgeCases/type-with-trailing-question-mark/)invalid-notnilable-missing-value.raml
  * tests/raml-1.0/EdgeCases/type-with-trailing-question-mark/ [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/EdgeCases/type-with-trailing-question-mark/)valid-nilables-with-values.raml
  * tests/raml-1.0/EdgeCases/type-with-trailing-question-mark/valid.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/EdgeCases/type-with-trailing-question-mark/valid.raml)
* https://github.com/raml-org/raml-spec/issues/707:
  * tests/raml-1.0/Methods/method-question-mark/valid.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Methods/method-question-mark/valid.raml)
* https://github.com/raml-org/raml-spec/issues/720:
  * tests/raml-1.0/Types/include-json-schema-element/invalid-pointer-nowhere.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/include-json-schema-element/invalid-pointer-nowhere.raml)
  * tests/raml-1.0/Types/include-json-schema-element/valid.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Types/include-json-schema-element/valid.raml)
* https://github.com/raml-org/raml-spec/issues/724:
  * tests/raml-1.0/Traits/is-node-format/invalid-is-single-value.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Traits/is-node-format/invalid-is-single-value.raml)
  * tests/raml-1.0/Traits/is-node-format/valid-is-array.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Traits/is-node-format/valid-is-array.raml)
* https://github.com/raml-org/raml-spec/issues/730:
  * tests/raml-1.0/Libraries/used-by-resourcetype/invalid-fragment-param-in-uses.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-resourcetype/invalid-fragment-param-in-uses.raml)
  * tests/raml-1.0/Libraries/used-by-resourcetype/invalid-spec-param-in-uses.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-resourcetype/invalid-spec-param-in-uses.raml)
  * tests/raml-1.0/Libraries/used-by-resourcetype/valid-spec.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-resourcetype/valid-spec.raml)
  * tests/raml-1.0/Libraries/used-by-trait/invalid-fragment-param-in-uses.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-trait/invalid-fragment-param-in-uses.raml)
  * tests/raml-1.0/Libraries/used-by-trait/invalid-spec-param-in-uses.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Libraries/used-by-trait/invalid-spec-param-in-uses.raml)
  * tests/raml-1.0/ResourceTypes/include-parameter/invalid-param-in-include-path.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/ResourceTypes/include-parameter/invalid-param-in-include-path.raml)
  * tests/raml-1.0/Traits/with-params/example.json [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Traits/with-params/example.json)
  * tests/raml-1.0/Traits/with-params/invalid-param-in-include-path.raml [link](https://github.com/raml-org/raml-tck/blob/i130_add_raml101_tests/tests/raml-1.0/Traits/with-params/invalid-param-in-include-path.raml)


Notes:
* https://github.com/raml-org/raml-spec/issues/707: Already covered by existing tests. Added more test though.
* https://github.com/raml-org/raml-spec/issues/716: No tests to add here. It's just a change of wording.
* https://github.com/raml-org/raml-spec/issues/725: Nothing to test here.
* https://github.com/raml-org/raml-spec/issues/727: Already covered in `tests/raml-1.0/Fragments/`
* https://github.com/raml-org/raml-spec/issues/728: Already covered by existing tests.
